src/model/engines/engine_base.nit

```diff
@@ -0,0 +1,59 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Base for code runners of a kind
+module engine_base
+
+import submissions
+
+# Any class capable of running some code in a defined language
+class Engine
+	# Which language is supported by the engine?
+	fun language: String is abstract
+
+	# Compile and run the tests for `program`
+	fun run(program: Program) do
```

Comment:
Is it really useful ? The client will only call run once and will have to check the updated `program` anyway to produce something to return
I can still add it, but is it really worth it?